src/renderer/components/+preferences/preferences-navigation/preferences-navigation.tsx

```diff
@@ -13,30 +13,58 @@ import type {
 import preferenceNavigationItemsInjectable from "./preference-navigation-items.injectable";
 
 import { observer } from "mobx-react";
+import { Icon } from "../../icon";
 
 interface Dependencies {
   navigationItems: IComputedValue<PreferenceNavigationItem[]>;
 }
 
 const NonInjectedPreferencesNavigation = ({
   navigationItems,
-}: Dependencies) => (
-  <Tabs
-    className="flex column"
-    scrollable={false}
-    onChange={(item: PreferenceNavigationItem) => item.navigate()}
-  >
-    <div className="header">Preferences</div>
-
-    {navigationItems.get().map((item) => (
+}: Dependencies) => {
+  const generalNavItems = navigationItems.get().filter(item => !item.fromExtension);
+  const extensionNavItems = navigationItems.get().filter(item => item.fromExtension);
```

Comment:
At least, split to two separate injectables. However, in the future real dependency is the array that contains hierarchy of navigation items (and the headers, icons etc.)